hazelcast/src/main/java/com/hazelcast/internal/config/MemberDomConfigProcessor.java

```diff
@@ -1682,7 +1682,11 @@ void handleReplicatedMapNode(Node node, final ReplicatedMapConfig replicatedMapC
 
     protected void handleMap(Node parentNode) throws Exception {
         String name = getAttribute(parentNode, "name");
-        MapConfig mapConfig = config.getMapConfig(name);
+        MapConfig mapConfig = config.getMapConfigs().get(name);
+        if (mapConfig == null) {
+            mapConfig = new MapConfig(name);
+            config.addMapConfig(mapConfig);
+        }
```

Comment:
Great finding, congrats :tada: We have `ConfigUtils#getConfig()` for a very similar concern. It uses a config pattern matcher though. We can reuse/duplicate that utility method for exact matching for a more convenient fix. WDYT?

Reply:
Something like this? https://github.com/hazelcast/hazelcast/pull/18187/files#diff-50dbca5dc4eb06e68d3442fcc744b0f3dc0c99e38dfd486759cfa0b42e43d1b0R136-R154

Reply:
Magnifico :ok_hand: :+1: :heart: